lib/TH/generic/THStorage.c

```diff
@@ -107,9 +107,9 @@ void THStorage_(free)(THStorage *storage)
   if(!storage)
     return;
 
-  if((storage->flag & TH_STORAGE_REFCOUNTED) && (storage->refcount > 0))
+  if((storage->flag & TH_STORAGE_REFCOUNTED) && (THAtomicGet(&storage->refcount) > 0))
   {
```

Comment:
it is because certain storages used to have negative refcount, and untouched somehow (before i added the flag about refcount). i suppose i could remove that, not sure for backward reasons.